git/gitattr/tree.go

```diff
@@ -114,18 +114,50 @@ func linesInTree(db *gitobj.ObjectDatabase, t *gitobj.Tree, mp *MacroProcessor)
 // if there are relevant .gitattributes matching that path.
 func (t *Tree) Applied(to string) []*Attr {
 	var attrs []*Attr
-	for _, line := range t.Lines {
-		if l, ok := line.(PatternLine); ok {
-			if l.Pattern().Match(to) {
-				attrs = append(attrs, line.Attrs()...)
-			}
+	if !t.MP.didReadMacros {
+		if t.SystemAttributes != nil {
+			t.MP.ProcessLines(t.SystemAttributes.Lines, true)
+		}
+		if t.UserAttributes != nil {
+			t.MP.ProcessLines(t.UserAttributes.Lines, true)
+		}
+		t.MP.ProcessLines(t.Lines, true)
+
+		if t.RepoAttributes != nil {
+			t.MP.ProcessLines(t.RepoAttributes.Lines, true)
+		}
+	}
```

Comment:
IIUC, the purpose of these steps is to load all the macros found in the various `gitattributes(5)` files (in the appropriate order) using just the `MacroLine` [case](https://github.com/git-lfs/git-lfs/blob/071e19e8ea03b1e40b181706909fb8c18d928e29/git/gitattr/macro.go#L62-L65) in the `ProcessLines()` method.

Since we don't care about the other lines in the files at this stage, the `PatternLine` [case](https://github.com/git-lfs/git-lfs/blob/071e19e8ea03b1e40b181706909fb8c18d928e29/git/gitattr/macro.go#L30-L61) is overhead which might be skipped.  Perhaps we could substitute for the whole `ProcessLines()` method a simpler `ProcessMacros()` method which only looked for macro definitions and ignored all other lines, and then use that new method here?